nav2_localization/include/nav2_localization/interfaces/sample_motion_model_base.hpp

```diff
@@ -0,0 +1,50 @@
+#ifndef NAV2_LOCALIZATION__SAMPLE_MOTION_MODEL_BASE_HPP_
+#define NAV2_LOCALIZATION__SAMPLE_MOTION_MODEL_BASE_HPP_
+
+#include "geometry_msgs/msg/pose.hpp"
+#include "nav_msgs/msg/odometry.hpp"
+
+namespace nav2_localization_base
+{
+
+/**
+ * @class SampleMotionModel
+ * @brief Abstract interface for sample motion model to adhere to with pluginlib
+ */
+class SampleMotionModel
```

Comment:
> Why not just MotionModel? Curious

To differentiate it from a class that returns a probability distribution across the entire pose space, rather than just return the most likely pose.

Reply:
Why not make this take in some probability weights? If an application doesn't want it, they can just set the noise values to 0